### ISSUE TITLE
no directory auto-complete  env variables with command

### ISSUE THREAD

**Issue by The-Alchemist** [fish-shell/fish-shell#2066]
It appears there's no auto-complete when using the example at http://fishshell.com/docs/current/faq.html#faq-single-env

e.g., in bash
```
JAVA_HOME=/Library/<TAB>
```

...works!  But in fish, this doesn't work:
```
env JAVA_HOME=/Library/<TAB>
```

Not sure if this is a bug or new feature request.

Thanks!

**Comment by zanchey** [comment]
This gets you most of the way there:

```
complete -c env -n "commandline -ct | grep -qF =" -f -a "(commandline -ct | cut -f 1 -d=)=(__fish_complete_path (commandline -ct | cut -f 2 -d'='))" --description "Redefine variable"
```

but I'm not convinced it's right.

**Comment by ridiculousfish** [comment]
We ought to do path completions after = signs.

**Comment by ridiculousfish** [comment]
This should be fixed as 85dad6913e77d81730f3aea478695429f441ebd5. Thanks for filing this!
